mentoring/programs/mentoring-lead.md

```diff
@@ -0,0 +1,58 @@
+# Internship Mentoring Lead Handbook
+
+## Overview
+
+The Mentoring Lead role is created to help upstream Kubernetes contributors to participates in various mentorship programs like [GSoC](https://github.com/kubernetes/community/blob/master/mentoring/programs/google-summer-of-code.md), [GSoD](https://github.com/kubernetes/community/blob/master/mentoring/programs/google-season-of-docs.md), [LFX Mentorship](https://github.com/kubernetes/community/blob/master/mentoring/programs/lfx-mentorship.md) and many more as an independent organisation or under Cloud Native Computing Foundation(CNCF). 
+
+## Responsibilities
+ The role lead should:
+
+- Keep a calendar of dates for each program (when to submit our project name, when to submit the project scope, when selecting period is, etc) throughout the   year and
```

Comment:
not for this PR - but we should look into how we can publish these dates on a public calendar so anyone can subscribe to it. Might help applicants as well.